hil/ext/switches/nexus.py

```diff
@@ -124,12 +124,22 @@ def disable_native(self, vlan_id):
     @staticmethod
     def connect(switch):
         """Connect to the switch."""
+        alternatives = ['[Pp]assword: ', '>', '#']
         console = pexpect.spawn(
             'ssh ' + switch.username + '@' + switch.hostname)
 
-        console.expect('Password: ')
-        console.sendline(switch.password)
+        outcome = console.expect(alternatives)
+        if outcome == 0:
+            console.sendline(switch.password)
+            outcome = console.expect(alternatives)
+        if outcome == 1:
+            console.sendline('en')
 
+        logger.debug('Logged in to switch %r', switch)
+
+        # send a new line so that we can "expect" a prompt again if we already
+        # matched when outcome was 2 (pub key auth)
+        console.sendline('')
```

Comment:
Why does this work here and not above?

Reply:
I actually don't know, which is why I put the fixme there (showing the output if we just send a newline). I was hoping you could provide some insights into why this doesn't work on the powerconnect (or some way of debugging this).

Sidenote: we send the Dell n3000 a [bunch of commands](https://github.com/CCI-MOC/hil/blob/master/hil/ext/switches/n3000.py#L117-L125) when connecting to it before getting the prompts, and that works too.

Reply:
Not sure. We should at least add some comments mentioning the difference.

Reply:
Sorry but I don't know what's not clear here. I have added the comment here (Line 140, 141) and in the dell file (Line 98-102) too which describes what happens if we just send a newline.